base/array.jl

```diff
@@ -1610,7 +1610,7 @@ function findnext(A, start::Integer)
     while i <= l
         a = A[i]
         if !warned && !(a isa Bool)
-            depwarn("In the future `findnext` will only work on boolean collections. Use `findnext(x->x!=0, A)` instead.", :findnext)
+            depwarn("In the future `findnext` will only work on boolean collections. Use `findnext(x->x!=0, A, start)` instead.", :findnext)
```

Comment:
This should probably be recommending `findnext(!iszero, A, start)`. Any particular reason why that wasn't done initially, @JeffBezanson? (You did this initial deprecation, right?)

Reply:
That was in https://github.com/JuliaLang/julia/pull/23812. I think the reason is that `iszero` throws an error for some types (see https://github.com/JuliaLang/julia/issues/23005), so the only replacement which is guaranteed to work is `x != 0`. But in most cases `iszero` is OK (it's the recommended deprecation for `Number`).

Reply:
Okay, thanks